flepimop/gempyor_pkg/src/gempyor/_pydantic_ext.py

```diff
@@ -180,3 +184,64 @@ def _read_and_validate_dataframe(
             model = RootModel[list[model]]
         data = [r.model_dump() for r in model.model_validate(data).root]
     return pd.DataFrame.from_records(data)
+
+
+@overload
+def _evaled_expression(val: str, *, target_type: Type[int]) -> int: ...
+@overload
+def _evaled_expression(val: Any, *, target_type: Type[int]) -> int | Any: ...
+
+
+@overload
+def _evaled_expression(val: str, *, target_type: Type[float]) -> float: ...
+@overload
+def _evaled_expression(val: Any, *, target_type: Type[float]) -> float | Any: ...
+
+
+def _evaled_expression(val: Any, *, target_type: Type[EE]) -> EE | Any:
```

Comment:
```suggestion
@overload
def _evaled_expression(val: EE | str, *, target_type: Type[EE]) -> EE: ...

@overload
def _evaled_expression(val: T, *, target_type: Type[EE]) -> T: ...

def _evaled_expression(val: EE | str | Any, *, target_type: Type[EE]) -> EE | Any:
```
This should simplify the overloads and convey that when given `EE | str` the return is `EE` and otherwise it is as given.

Reply:
Also, is there a particular reason why the `target_type` is a keyword only argument? Seems like it just adds more typing for an internal function that has good documentation.

Reply:
Okay sounds good, happy to get rid of `*`. I thought it would be a good clarity thing, to delineate between the target and what it returned (if they were not the same), but this would just become clear when an error gets thrown.